pkg/kubernetes/watch/apps.go

```diff
@@ -27,3 +27,7 @@ func DaemonSets(client Provider[*appsv1.DaemonSetList]) *Watcher[appsv1.DaemonSe
 func Deployments(client Provider[*appsv1.DeploymentList]) *Watcher[appsv1.Deployment] {
 	return FromClient[*appsv1.DeploymentList, appsv1.Deployment](client)
 }
+
+func StatefulSet(client Provider[*appsv1.StatefulSetList]) *Watcher[appsv1.StatefulSet] {
```

Comment:
Should be plural, I'd say.

```suggestion
func StatefulSets(client Provider[*appsv1.StatefulSetList]) *Watcher[appsv1.StatefulSet] {
```